tensorflow/compiler/xla/service/gpu/gemm_algorithm_picker.cc

```diff
@@ -220,49 +426,59 @@ static StatusOr<absl::optional<se::blas::AlgorithmType>> DoUncachedGemmAutotune(
 static StatusOr<absl::optional<se::blas::AlgorithmType>> DoGemmAutotune(
     const HloInstruction* instr, const GemmBackendConfig& gemm_config,
     se::DeviceMemoryAllocator* allocator, se::Stream* stream) {
+  VLOG(3) << "Starting autotune of GemmThunk " << instr->ToString();
   const HloInstruction* lhs = instr->operand(0);
   const HloInstruction* rhs = instr->operand(1);
 
+  GpuGemmConfig config = GetGpuGemmConfig(instr);
+  PrimitiveType element_type = config.output_shape.element_type();
   // Don't run autotuning concurrently on the same GPU.
   absl::MutexLock gpu_lock(&GetGpuMutex(stream->parent()));
 
-  GemmCacheKey key =
-      std::make_tuple(stream->parent(), lhs->shape(), rhs->shape(),
-                      instr->shape(), gemm_config.SerializeAsString());
+  if (stream->parent()->SupportsBlasPlans() && config.use_cublaslt &&
```

Comment:
Maybe slightly pedantic, but we ought to check the flag first, lest `SupportBlasPlans` has some kind of side effect.

Reply:
`SupportsBlasPlans()` is introduced in this PR and has no side effects. It returns true if the CUDA version is greater than or equal to 11, and false otherwise.